x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/DatafeedJobsRestIT.java

```diff
@@ -128,6 +127,11 @@ private void addAirlineData() throws IOException {
                 + "          \"keyword\":{\"type\":\"keyword\"}"
                 + "         }"
                 + "       },"
+                + "      \"airline_lowercase_rt\": { "
+                + "        \"type\":\"runtime\","
+                + "        \"runtime_type\": \"keyword\","
+                + "        \"script\" : { \"source\": \"emit(params._source.airline.toLowerCase())\" }"
```

Comment:
thanks for writing this test. One point is, there is no need to go to _source to get this field as it's indexed as keyword and has doc_values. The most appropriate way to do this would then be `emit(doc['airline'].toLowerCase())`

I care about it because _source is available for runtime fields, yet it is still not the recommended way to retrieve values, which we plan to address over time.

Reply:
What you can't see in the diff is that `airline` is actually a text multi field with a keyword sub field

```
      "airline" : { 
        "type": "text",
        "fields": {
          "keyword" : { 
            "type": "keyword"
          }
        }
      },
```

But I do see the sense in what you're suggesting so I changed the script to:
```emit(doc['airline.keyword'].toLowerCase())```

Unfortunately that failed with the error 
```
          "caused_by" : {
            "type" : "illegal_argument_exception",
            "reason" : "dynamic method [org.elasticsearch.index.fielddata.ScriptDocValues.Strings, toLowerCase/0] not found"
          }
```

I had to put a `toString()` in to get it to work. 
```emit(doc['airline.keyword'].toString().toLowerCase())```

Is this a regression as that example worked previously? 

BTW I'm testing this simply by asking for the docvalue_field in the search
```
GET farequote/_search
{
  "docvalue_fields": ["airline_lowercase"]
}
```

Reply:
nice, I thought I checked out the test data and that field was keyword, but close enough. Ping me if you notice anything when playing with runtime fields ;)

Reply:
I think I forgot a .value possibly? From your message I did not get if you managed to make it work after all, if not ping me and let's make sure that it works on doc_values

Reply:
👍 it works with `.value` thanks